src/community/features-templating/features-templating-core/src/main/java/org/geoserver/featurestemplating/writers/XHTMLTemplateWriter.java

```diff
@@ -48,13 +62,41 @@ public void startTemplateOutput(EncodingHints encodingHints) throws IOException
             List style = encodingHints.get(STYLE, List.class);
             if (style != null) encodeHeadContent(STYLE, style);
             encodeLinks(encodingHints);
+            injectJSONLD(encodingHints);
             streamWriter.writeEndElement();
             streamWriter.writeStartElement("body");
-        } catch (XMLStreamException e) {
+        } catch (Exception e) {
             throw new IOException(e);
         }
     }
 
+    private void injectJSONLD(EncodingHints encodingHints)
+            throws ExecutionException, IOException, XMLStreamException {
+        List injectJSONLD = encodingHints.get(JSON_LD_SCRIPT, List.class, Collections.emptyList());
+        if (!injectJSONLD.isEmpty()) {
+            List<FeatureCollection> collections = (List<FeatureCollection>) injectJSONLD;
+            JSONLDOutputHelper jsonldOutputHelper = new JSONLDOutputHelper();
+            EncodingHints jsonLdHints = jsonldOutputHelper.optionsToEncodingHints(collections);
+            streamWriter.writeStartElement("script");
+            streamWriter.writeAttribute("type", "application/ld+json");
+            ByteArrayOutputStream baos = null;
+            try (JSONLDWriter writer = getJSONLDWriter(baos = new ByteArrayOutputStream())) {
```

Comment:
I guess the output field could have been used here, but it did not in the end.

Reply:
fixed.